WordPress/src/main/java/org/wordpress/android/ui/prefs/timezone/SiteSettingsTimezoneViewModel.kt

```diff
@@ -0,0 +1,211 @@
+            val jsonTimezonesByContinent = jsonResponse.getJSONObject("timezones_by_continent")
+            val jsonTimezonesManualOffsets = jsonResponse.getJSONArray("manual_utc_offsets")
+
+            Continents.values().map {
+                timezonesList.add(TimezoneHeader(it.s))
+                addTimezoneItems(jsonTimezonesByContinent.getJSONArray(it.s))
+            }
+
+            timezonesList.add(TimezoneHeader("Manual Offsets"))
+            addManualTimezoneItems(jsonTimezonesManualOffsets)
+
+            AppLog.d(SETTINGS, timezonesList.toString())
+            _timezones.postValue(timezonesList)
+        } catch (e: JSONException) {
+            AppLog.e(SETTINGS, "Error parsing timezones", e)
+            _dismiss.postValue(Unit)
+        }
+    }
+
+    private fun addTimezoneItems(jsonTimezones: JSONArray) {
+        for (i in 0 until jsonTimezones.length()) {
+            val timezone = jsonTimezones.getJSONObject(i)
+            val city = timezone.getString("label")
+            val zone = timezone.getString("value")
+            val zoneOffset = getZoneOffset(zone)
+            val zoneTime = getTimeAtZone(zone)
+            timezonesList.add(TimezoneItem(city, zone, zoneOffset, zoneTime))
+        }
+    }
+
+    private fun addManualTimezoneItems(jsonTimezones: JSONArray) {
+        for (i in 0 until jsonTimezones.length()) {
+            val timezone = jsonTimezones.getJSONObject(i)
+            timezonesList.add(TimezoneItem(timezone.getString("label"), timezone.getString("value")))
+        }
+    }
+
+    private fun getZoneOffset(zone: String): String {
+        val zoneId = getZoneId(zone)
+
+        return if (zoneId != null) {
+            val offset = ZonedDateTime.now(zoneId).offset
+            getZoneDisplayName(zoneId, offset)
+        } else {
+            ""
+        }
+    }
+
+    private fun getZoneDisplayName(zoneId: ZoneId, offset: ZoneOffset): String {
+        val zoneDisplayName = zoneId.getDisplayName(FULL, Locale.getDefault())
+        val zone =  zoneDisplayName.substringBefore("/")
+
+        val zoneDisplay = if (zone.isNotBlank()) zone else zoneDisplayName
+        val offsetDisplay = if (offset.id == "Z") "" else "(GMT$offset)"
+
+        return "$zoneDisplay $offsetDisplay"
+    }
+
+    private fun getTimeAtZone(zone: String): String {
+        val dateTimeFormatter = DateTimeFormatter.ofPattern("h:mm a")
+        val zoneId = getZoneId(zone)
+
+        return if (zoneId != null) {
+            val zoneDateTime = ZonedDateTime.now(zoneId)
+            zoneDateTime.format(dateTimeFormatter)
+        } else  {
+            ""
+        }
+    }
+
+    private fun getZoneId(zone: String): ZoneId? {
+        return try {
+            ZoneId.of(zone)
+        } catch (e: ZoneRulesException) {
+            AppLog.e(SETTINGS, "Error parsing zoneId", e)
+            null
+        }
+    }
+
+    enum class Continents(val s: String) {
```

Comment:
For readability, we try to avoid ambitious attributes names like `s`. Also, it looks like the continent names from here will be visible in UI, so they need to be in strings resources.